scipy/spatial/tests/test_distance.py

```diff
@@ -1427,78 +1438,75 @@ def test_pdist_calling_conventions(self):
-                except Exception as e:
-                    e_cls = e.__class__
-                    if verbose > 2:
-                        print(e_cls.__name__)
-                        print(e)
-                    for new_type in test[1]:
-                        X2 = new_type(X1)
-                        assert_raises(e_cls, pdist, X2, metric=metric)
-                else:
-                    for new_type in test[1]:
-                        y2 = pdist(new_type(X1), metric=metric)
-                        _assert_within_tol(y1, y2, eps, verbose > 2)
-
-    def test_pdist_out(self):
+        for test in tests:
+            X1 = test[0][::5, ::2]
+            try:
+                y1 = pdist(X1, metric=metric)
+            except Exception as e:
+                e_cls = e.__class__
+                if verbose > 2:
+                    print(e_cls.__name__)
+                    print(e)
+                for new_type in test[1]:
+                    X2 = new_type(X1)
+                    assert_raises(e_cls, pdist, X2, metric=metric)
+            else:
+                for new_type in test[1]:
+                    y2 = pdist(new_type(X1), metric=metric)
+                    _assert_within_tol(y1, y2, eps, verbose > 2)
+
+    def test_pdist_out(self, metric):
         # Test that out parameter works properly
         eps = 1e-07
         X = eo['random-float32-data'][::5, ::2]
         out_size = int((X.shape[0] * (X.shape[0] - 1)) / 2)
-        for metric in _METRICS_NAMES:
-            kwargs = dict()
-            if metric == 'minkowski':
-                kwargs['p'] = 1.23
-            out1 = np.empty(out_size, dtype=np.double)
-            Y_right = pdist(X, metric, **kwargs)
-            Y_test1 = pdist(X, metric, out=out1, **kwargs)
-            # test that output is numerically equivalent
-            _assert_within_tol(Y_test1, Y_right, eps)
-            # test that Y_test1 and out1 are the same object
-            assert_(Y_test1 is out1)
-            # test for incorrect shape
-            out2 = np.empty(out_size + 3, dtype=np.double)
-            assert_raises(ValueError, pdist, X, metric, out=out2, **kwargs)
-            # test for (C-)contiguous output
-            out3 = np.empty(2 * out_size, dtype=np.double)[::2]
-            assert_raises(ValueError, pdist, X, metric, out=out3, **kwargs)
-            # test for incorrect dtype
-            out5 = np.empty(out_size, dtype=np.int64)
-            assert_raises(ValueError, pdist, X, metric, out=out5, **kwargs)
-
-    def test_striding(self):
+        kwargs = dict()
+        if metric == 'minkowski':
+            kwargs['p'] = 1.23
+        out1 = np.empty(out_size, dtype=np.double)
+        Y_right = pdist(X, metric, **kwargs)
+        Y_test1 = pdist(X, metric, out=out1, **kwargs)
+        # test that output is numerically equivalent
+        _assert_within_tol(Y_test1, Y_right, eps)
+        # test that Y_test1 and out1 are the same object
+        assert_(Y_test1 is out1)
```

Comment:
Use plain `assert`

Reply:
Plain asserts are also not exactly the height of modern test suites, but it might still be better than `assert_()`

Reply:
What do you mean?

Reply:
See e.g. https://pypi.org/project/flake8-assert-msg/.

Reply:
Thanks I didn't know this hook. This is not directed to tests but when you have assertions in the code itself to explain them. Pytest don't say anything about adding messages and tests should be self explanatory as stated in the doc (they also advise not to write docstrings).

Reply:
Fair enough. I guess I just prefer more functional statements like `np.testing.assert_equal` than the statement-syntax of `assert`, but they do have their use of course.

Reply:
I agree that it's not very natural also because running `python -O` removes assertions.